sdgym/sdgym_result_explorer/result_handler.py

```diff
@@ -32,6 +37,106 @@ def load_synthetic_data(self, file_path):
+        pass
+
+    def _compute_wins(self, result):
+        synthesizers = result['Synthesizer'].unique()
+        datasets = result['Dataset'].unique()
+        result['Win'] = 0
+        for dataset in datasets:
+            score_baseline = result.loc[
+                (result['Synthesizer'] == SYNTHESIZER_BASELINE) & (result['Dataset'] == dataset)
+            ]['Quality_Score'].to_numpy()
+            if score_baseline.size == 0:
+                continue
+
+            for synthesizer in synthesizers:
+                loc_synthesizer = (result['Synthesizer'] == synthesizer) & (
+                    result['Dataset'] == dataset
+                )
+                score_synthesizer = result.loc[loc_synthesizer]['Quality_Score'].to_numpy()
+                result.loc[loc_synthesizer, 'Win'] = (score_synthesizer > score_baseline).astype(
+                    int
+                )
+
+    def _get_summarize_table(self, folder_to_results, folder_infos):
+        """Create a summary table from the results."""
+        columns = []
+        for folder, results in folder_to_results.items():
+            date_str = folder_infos[folder]['date']
+            date_obj = datetime.strptime(date_str, '%m_%d_%Y')
+            column_name = (
+                f'{date_str}'
+                f' - # datasets: {folder_infos[folder]["# datasets"]}'
+                f' - sdgym version: {folder_infos[folder]["sdgym_version"]}'
+            )
+            results = results.loc[results['Synthesizer'] != SYNTHESIZER_BASELINE]
+            column_data = results.groupby(['Synthesizer'])['Win'].sum()
+            columns.append((date_obj, column_name, column_data))
+
+        columns.sort(key=operator.itemgetter(0))
+        summarized_results = pd.DataFrame()
+        for _, column_name, column_data in reversed(columns):
+            summarized_results[column_name] = column_data
+
+        summarized_results = summarized_results.fillna('-')
+        return summarized_results
+
+    def _get_column_name_infos(self, folder_to_results):
+        folder_to_info = {}
+        for folder, results in folder_to_results.items():
+            yaml_files = self._get_results_files(folder, prefix='run_', suffix='.yaml')
+            if not yaml_files:
+                continue
+
+            run_id_info = self._load_yaml_file(folder, yaml_files[0])
+            num_datasets = results.loc[
+                results['Synthesizer'] == SYNTHESIZER_BASELINE, 'Dataset'
+            ].nunique()
+            folder_to_info[folder] = {
+                'date': run_id_info.get('starting_date')[:10],
+                'sdgym_version': run_id_info.get('sdgym_version'),
+                '# datasets': num_datasets,
+            }
+
+        return folder_to_info
+
+    def summarize(self, folder_name):
+        """Summarize the results in the specified folder."""
+        all_folder = [f for f in self.list() if f.startswith(folder_name[:-11])]
```

Comment:
can we make the prefix a constant and use that instead of magic numbers like `folder_name[:-11]`